### PR TITLE
Remove all compiler warnings

### PR DESCRIPTION
**Pull request by markschaake**
This PR removes all compiler warnings and requires going forward that no more warnings are allowed by adding `-Xfatal-warnings` to the build's `scalacOptions`.

This PR was motivated by #76, though it does not fix the bug. I am trying to reproduce #76 via a unit test but so far have not been able to. However, please consider merging these small changes to prevent compiler warnings being introduced in the future.

### CODE REVIEW

File: avro4s-core/src/test/scala/com/sksamuel/avro4s/AvroInputStreamTest.scala [hunk 1]
```diff
@@ -2,13 +2,12 @@ package com.sksamuel.avro4s
...
 import java.util.UUID
-
-import org.scalatest.concurrent.Timeouts
+import org.scalatest.concurrent.TimeLimits
 import org.scalatest.{Matchers, WordSpec}
 
 import shapeless.{:+:, Coproduct, CNil}
 
-class AvroInputStreamTest extends WordSpec with Matchers with Timeouts {
+class AvroInputStreamTest extends WordSpec with Matchers with TimeLimits {
```
Comment (markschaake) [inline avro4s-core/src/test/scala/com/sksamuel/avro4s/AvroInputStreamTest.scala]: `Timeoutes` was deprectated in favor of `TimeLimits`

File: avro4s-json/src/main/scala/com/sksamuel/avro4s/json/JsonToAvroConverter.scala [hunk 2]
```diff
@@ -38,7 +38,9 @@ class JsonToAvroConverter(namespace: String, avroStringTypeIsString: Boolean = f
     case JString(_) => createStringSchema
     case JObject(values) =>
       val record = Schema.createRecord(name, null, namespace, false)
-      val fields = values.map { case (name, value) => new Schema.Field(name, convert(name, value), null, null) }
+      val doc: String = null
+      val default: AnyRef = null
+      val fields = values.map { case (name, value) => new Schema.Field(name, convert(name, value), doc, default) }
```
Comment (markschaake) [inline avro4s-json/src/main/scala/com/sksamuel/avro4s/json/JsonToAvroConverter.scala]: Some `Schema.Field` constructors were deprecated. Providing explicit types for the arguments ensures the correct non-deprecated constructor is called.

File: avro4s-macros/src/main/scala/com/sksamuel/avro4s/SchemaFor.scala [hunk 3]
```diff
@@ -370,28 +367,21 @@ object SchemaFor {
...
+    def toDefaultValue(value: Any): Any = value match {
+      case x: Int => x
+      case x: Long => x
+      case x: Boolean => x
+      case x: Double => x
+      case x: Seq[_] => x.asJava
+      case x: Map[_, _] => x.asJava
+      case Some(x) => x
+      case None => JsonProperties.NULL_VALUE
+      case _ => value.toString
```
Comment (markschaake) [inline avro4s-macros/src/main/scala/com/sksamuel/avro4s/SchemaFor.scala]: This is the riskiest change in the PR. The `Schema.Field` constructors that took a `JsonNode` argument for the default value have all been deprecated. There was a unit test in place that covered the expected behavior here, and it still passes with this change.

**Comment by sksamuel** [comment]
:+1:

**Pull request merged**